aws/resource_aws_config_remediation_configuration.go

```diff
@@ -0,0 +1,253 @@
+		},
+	}
+}
+
+func expandConfigRemediationConfigurationParameters(configured *schema.Set) (map[string]*configservice.RemediationParameterValue, error) {
+	results := make(map[string]*configservice.RemediationParameterValue)
+
+	for _, item := range configured.List() {
+		detail := item.(map[string]interface{})
+		rpv := configservice.RemediationParameterValue{}
+		resourceName, ok := detail["name"].(string)
+		if ok {
+			results[resourceName] = &rpv
+		} else {
+			return nil, fmt.Errorf("Could not extract name from parameter.")
+		}
+		if resourceValue, ok := detail["resource_value"].(string); ok && len(resourceValue) > 0 {
+			rpv.ResourceValue = &configservice.ResourceValue{
+				Value: &resourceValue,
+			}
+		} else if staticValue, ok := detail["static_value"].(string); ok && len(staticValue) > 0 {
+			rpv.StaticValue = &configservice.StaticValue{
+				Values: []*string{&staticValue},
+			}
+		} else {
+			return nil, fmt.Errorf("Parameter '%s' needs one of resource_value or static_value", resourceName)
+		}
+	}
+
+	return results, nil
+}
+
+func flattenRemediationConfigurationParameters(parameters map[string]*configservice.RemediationParameterValue) []interface{} {
+	var items []interface{}
+
+	for key, value := range parameters {
+		item := make(map[string]interface{})
+		item["name"] = key
+		if value.ResourceValue != nil {
+			item["resource_value"] = *value.ResourceValue.Value
+		}
+		if value.StaticValue != nil && len(value.StaticValue.Values) > 0 {
+			item["static_value"] = *value.StaticValue.Values[0]
+		}
+
+		items = append(items, item)
+	}
+
+	return items
+}
+
+func resourceAwsConfigRemediationConfigurationPut(d *schema.ResourceData, meta interface{}) error {
+	conn := meta.(*AWSClient).configconn
+
+	name := d.Get("config_rule_name").(string)
+	remediationConfigurationInput := configservice.RemediationConfiguration{
+		ConfigRuleName: aws.String(name),
+	}
+
+	if v, ok := d.GetOk("parameter"); ok {
+		params, err := expandConfigRemediationConfigurationParameters(v.(*schema.Set))
+		if err != nil {
+			return err
+		}
+		remediationConfigurationInput.Parameters = params
+	}
+	if v, ok := d.GetOk("resource_type"); ok {
+		remediationConfigurationInput.ResourceType = aws.String(v.(string))
+	}
+	if v, ok := d.GetOk("target_id"); ok && v.(string) != "" {
+		remediationConfigurationInput.TargetId = aws.String(v.(string))
+	}
+	if v, ok := d.GetOk("target_type"); ok && v.(string) != "" {
+		remediationConfigurationInput.TargetType = aws.String(v.(string))
+	}
+	if v, ok := d.GetOk("target_version"); ok && v.(string) != "" {
+		remediationConfigurationInput.TargetVersion = aws.String(v.(string))
+	}
+
+	input := configservice.PutRemediationConfigurationsInput{
+		RemediationConfigurations: []*configservice.RemediationConfiguration{&remediationConfigurationInput},
+	}
+	log.Printf("[DEBUG] Creating AWSConfig remediation configuration: %s", input)
+	_, err := conn.PutRemediationConfigurations(&input)
+	if err != nil {
+		return err
```

Comment:
In our error returns, we prefer wrapping the AWS API error with some context. For example
```suggestion
		return fmt.Errorf("Failed to create AWSConfig remediation configuration: %w", err)
```